common-custom-user-data-gradle-plugin/src/main/java/com/gradle/CustomBuildScanConfig.java

```diff
@@ -0,0 +1,387 @@
+package com.gradle;
+
+import com.gradle.scan.plugin.BuildScanExtension;
+import org.gradle.api.Project;
+import org.gradle.api.invocation.Gradle;
+import org.gradle.api.tasks.testing.Test;
+
+import java.io.BufferedReader;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.Reader;
+import java.io.UnsupportedEncodingException;
+import java.net.URLEncoder;
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+import java.util.Properties;
+import java.util.concurrent.TimeUnit;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static com.gradle.Hasher.hashValue;
+
+final class CustomBuildScanConfig {
+
+    static void configureBuildScan(BuildScanExtension buildScan, Gradle gradle) {
+        tagOs(buildScan);
+        tagIde(buildScan, gradle);
+        tagCiOrLocal(buildScan);
+        addCiMetadata(buildScan);
+        addGitMetadata(buildScan);
+        captureTestProperties(buildScan, gradle);
```

Comment:
We should have a way of disabling `captureTestProperties` it can be spammy in some builds.

Reply:
I suggest to wait with adding an extension until we have a first version ready to ship.

Reply:
Yeah, I implemented this in a branch already, but using a simple system property instead of an extension. This will make it easier to enable/disable any optional build scan features without changing the actual build. These changes can be made persistent via `gradle.properties`.

I will submit this as a separate PR, which also includes functionality to optionally capture the output of `ps` and `jps` as custom values.